docs/source/developing/ci.adoc

```diff
@@ -10,21 +10,45 @@ toc::[]
 [[ci-overview]]
 == Overview
 
-The following sections describes the various CI setups Minishift utilizes.
+Minishift uses the following CI build servers: CentOS, AppVeyor, Circle, and Travis. These servers build incoming pull requests and provide the status of the builds to ensure continuous integration. Except for Travis CI, they also archive build artifacts of successful builds.
 
-[[ci-setup]]
+image::<CI_Servers>[CI Servers]
+
+The following sections describe the various CI build servers Minishift utilizes.
+
+[[centos-ci]]
 == CentOS CI
 
-Minishift uses several CI build servers, including https://ci.centos.org/[CentOS CI]. The server
-builds incoming pull requests and pushes to master. It also archives build artifacts of successful
-builds.
+https://ci.centos.org/[CentOS CI] server is the only CI server which can run integration tests, as it supports nested virtualization. Thus it is used to run integration tests as well as unit tests and code format tests. It also verifies the cross compilation of the binary.
```

Comment:
I really dislike "code format tests". These are no tests. These are checks - "code style checks". A test implies some sort of sampling. Here are known rules about the formatting are verified.